corehq/apps/products/views.py

```diff
@@ -113,51 +118,63 @@ def page_context(self):
-
-    def get_archive_text(self, is_archived):
-        if is_archived:
-            return _("This will re-activate the product, and the product will show up in reports again.")
-        return _("As a result of archiving, this product will no longer appear in reports. "
-                 "This action is reversable; you can reactivate this product by viewing "
-                 "Show Archived Products and clicking 'Unarchive'.")
-
     @property
     def product_data(self):
-        data = []
-        if self.show_inactive:
-            products = Product.archived_by_domain(
-                domain=self.domain,
-                limit=self.limit,
-                skip=self.skip(),
-            )
-        else:
-            products = Product.by_domain(
-                domain=self.domain,
-                limit=self.limit,
-                skip=self.skip(),
-            )
+        start = (self.page - 1) * self.limit
+        end = start + self.limit
+        return map(self.make_product_dict, self.product_queryset[start:end])
 
-        for p in products:
-            if p.program_id:
-                program = Program.get(p.program_id)
-            else:
-                program = get_or_create_default_program(self.domain)
-                p.program_id = program.get_id
-                p.save()
-
-            info = p._doc
-            info['program'] = program.name
-            info['edit_url'] = reverse('commtrack_product_edit', kwargs={'domain': self.domain, 'prod_id': p._id})
-            info['archive_action_desc'] = self.get_archive_text(self.show_inactive)
-            info['archive_action_text'] = _("Un-Archive") if self.show_inactive else _("Archive")
-            info['archive_url'] = reverse(
-                'unarchive_product' if self.show_inactive else 'archive_product',
-                kwargs={'domain': self.domain, 'prod_id': p._id}
-            )
-            data.append(info)
-        return data
+    def make_product_dict(self, product):
+        archive_config = self.get_archive_config()
+        return {
+            'name': product.name,
+            'product_id': product.product_id,
+            'code': product.code,
+            'unit': product.units,
+            'description': product.description,
+            'program': self.program_name(product),
+            'edit_url': reverse(
+                'commtrack_product_edit',
+                kwargs={'domain': self.domain, 'prod_id': product.product_id}
+            ),
+            'archive_action_desc': archive_config['archive_text'],
+            'archive_action_text': archive_config['archive_action'],
+            'archive_url': reverse(
+                archive_config['archive_url'],
+                kwargs={'domain': self.domain, 'prod_id': product.product_id}
+            ),
+        }
+
+    def program_name(self, product):
+        if product.program_id:
+            return Program.get(product.program_id).name
```

Comment:
I don't think this needs to block merge especially since it's not a change in behavior - but I do think it's worth fixing it soon. Let me know if you want me to merge first and you'll track elsewhere.